File: data/code_pairs/fortran/26.f90
module great_relations_mod
    implicit none
    private
    public :: check_relationship

contains

    function check_relationship(from, to) result(is_great)
        character(len=*), intent(in) :: from, to
        logical :: is_great

        is_great = .false.

        ! Define relationships
        if ((from == "i1" .and. to == "x1") .or. &
            (from == "c1" .and. to == "a1") .or. &
            (from == "a1" .and. to == "o1") .or. &
            ! Add all other relationships here as needed...
            (from == "z1" .and. to == "g1")) then
            is_great = .true.
        endif
    end function check_relationship

end module great_relations_mod

File: data/code_pairs/cpp/26.cpp
#ifndef GREATRELATIONS_H
#define GREATRELATIONS_H

#include <vector>
#include <string>

struct Relationship {
    std::string from;
    std::string to;

    Relationship(std::string f, std::string t) : from(f), to(t) {}
};

class GreatRelations {
private:
    std::vector<Relationship> relationships;

public:
    GreatRelations() {
        relationships = {
            {"i1", "x1"},
            {"c1", "a1"},
            {"a1", "o1"},
            {"j1", "aa1"},
            {"h1", "l1"},
            // Add all other relationships here...
            {"z1", "g1"}
        };
    }

    bool checkRelationship(const std::string& from, const std::string& to) const {
        for (const auto& rel : relationships) {
            if (rel.from == from && rel.to == to) {
                return true;
            }
        }
        return false;
    }
};

#endif // GREATRELATIONS_H